src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -236,20 +239,18 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 
 						}
 						if (parseResult != null) {
-							int startIndex = -1;
-							for (int i = 0; (startIndex = nextUnescaped(pattern, '%', startIndex + 1)) != -1; i++) {
-								int endIndex = nextUnescaped(pattern, '%', startIndex + 1);
+							assert parseResult.source != null; // parse results from parse_i have a source
+							List<TypePatternElement> types = parseResult.source.getElements(TypePatternElement.class);
```

Comment:
Do we have to iterate to find TypePatternElements here? Could they be referenced separately since they're special?